lib/vsc/mympirun/main.py

```diff
@@ -0,0 +1,124 @@
+    # We are using sys.argv because generaloption depends on the the returned scriptname
+    if optionparser.options.setmpi:
+        setmpi = optionparser.options.setmpi
+        optionparser.log.debug("mympirun has been forced to use %s as MPI flavor", setmpi)
+    else:
+        setmpi = sys.argv[0]
+        optionparser.log.debug("mympirun will be executed by %s", setmpi)
+
+    scriptname, mpi, found_mpi = what_mpi(setmpi, mpim.MPI)
+    optionparser.log.debug("Found MPI classes %s", found_mpi)
+    found_mpi_names = sorted([x.__name__ for x in found_mpi])
+
+    if optionparser.options.showmpi:
+        fancylogger.setLogLevelInfo()
+        optionparser.log.info("Found MPI classes %s", ", ".join(found_mpi_names))
+        return None
+
+    # Select a Scheduler from the available schedulers
+    sched_name = getattr(optionparser.options, 'setsched', None)
+    if sched_name is None:
+        # legacy naming
+        sched_name = getattr(optionparser.options, 'schedtype', None)
+
+    sched, found_sched = what_sched(sched_name, schedm)
+    optionparser.log.debug("Found Sched classes %s", found_sched)
+    found_sched_names = sorted([x.__name__ for x in found_sched])
+
+    if optionparser.options.showsched:
+        fancylogger.setLogLevelInfo()
+        optionparser.log.info("Found Sched classes %s", ", ".join(found_sched_names))
+        return None
+
+    if mpi is None:
+        optionparser.log.raiseException(("No MPI class found that supports scriptname %s; isfake %s). Please use "
+                                         "mympirun through one of the direct calls or make sure the mpirun command can"
+                                         " be found. Found MPI %s") %
+                                        (scriptname, isfake, ", ".join(found_mpi_names)))
+    else:
+        optionparser.log.debug("Found MPI class %s (scriptname %s; isfake %s)", mpi.__name__, scriptname, isfake)
+
+    if sched is None:
+        optionparser.log.raiseException("No sched class found (options.schedtype %s ; found Sched classes %s)",
+                                        sched_name, ", ".join(found_sched_names))
+    else:
+        optionparser.log.debug("Found sched class %s from options.schedtype %s (all Sched found %s)",
+                               sched.__name__, sched_name, ", ".join(found_sched_names))
+
+    if not optionparser.args and not getattr(optionparser.options, 'print_launcher', False):
+        optionparser.log.warn("no mpi script provided")
+        return None
+
+    return mpi, sched, optionparser
+
+
+def main(mpim, mpiopt, schedm):
+    """
+    Main function: mpim is the base mpi module (with the MPI class), mpiopt is the option parser class
+    """
+    try:
+        instance_options = get_mpi_and_sched_and_options(mpim, mpiopt, schedm)
+        if instance_options:
+            instance = getinstance(*instance_options)
+            instance.main()
+    except Exception as e:
```

Comment:
please use `err` rather than `e` (don't use single-letter variable names outside of list comprehensions)

Reply:
fixed